src/status_im/ui/screens/wallet/choose_recipient/views.cljs

```diff
@@ -61,7 +62,7 @@
                        ;:torchMode     (camera/set-torch camera-flashlight)
                        :onBarCodeRead #(when-not @read-once?
                                          (reset! read-once? true)
-                                         (re-frame/dispatch [:wallet/fill-request-from-url (camera/get-qr-code-data %) :qr]))}]]
+                                         (re-frame/dispatch [:wallet.send/resolve-ens-addresses (camera/get-qr-code-data %) :qr]))}]]
```

Comment:
ideally we want events to be named after what happened not an action, that is why fx/defn can take a vector or events in it metadata, so you can associate multiple events with a particular function. In that case it could be `::qr-code-request-scanned` for instance

Reply:
I think I follow.  I'm still not sure I understand how the `::event-name` notation would apply here since that appears to only let you auto-resolve names within the current namespace (not for example, `choose-recipients.core` since the `qr-code-request-scanned` event is being called from `choose-recipients.view`.  I've pushed a new commit that adds `wallet.send/qr-code-request-scanned` as another alias for `resolve-ens-addresses`.

Reply:
`:hello` is a regular keyword
`:hi/hello` is a namespaced keyword
`::hello` is a namespaced keyword, if you are in `status-im.hi` namespace it will actually be resolved to `:status-im.hi` namespace
`::hi/hello` is also a namespace keyword, if you are in a namespace where the `status-im.hi` namespace is required like this: `[status-im.hi :as hi]` then it is equivalent to `:status-im.hi/hello` anywhere or `::hello` in the `status-im.hi` namespace

so `::` is shorthand for `use current namespace as namespace for this keyword` or if the keyword as a namespace it is shorthand for `resolve this namespace this is an alias`. If the alias doesn't exist it's a compilation error

Reply:
So you register the event as `::event-name` and you use it as `::alias/event-name` in other namespaces. The advantage is that it helps preventing hidden circular dependencies because it forces to import the namespaces where the events are declared, as well as conflict in event names